viewer/packages/api/src/public/migration/Cognite3DViewer.ts

```diff
@@ -1252,7 +1257,10 @@ export class Cognite3DViewer {
     if (isVisible) {
       const camera = this.cameraManager.getCamera();
       TWEEN.update(time);
-      this.recalculateBoundingBox();
+      if (this._nearAndFarPlaneNeedsUpdate) {
```

Comment:
In general this should probably work but I am not sure how we should handle cases where custom objects are not static, i.e. bounding box can change without adding/removing any objects. Because with this approach, custom objects that won't fit inside initial model bounding box could be clipped during rendering. Besides that, I guess it can apply to the model itself because it can have some animations/interactions as you've probably seen in the docs.

Reply:
Probably, one approach is to expose bounding box recalculation method to the user or add a flag that will enable/disable this recalculation for each frame. And the last option is to go through custom objects every frame but save time from model's bounding box.